### PR TITLE
Ampers: Kirsten

### PR DESCRIPTION
**Pull request by kschumy**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I had so much trouble with this. I could not decide what to do with rooms. I went back and forth about whether to make this a separate class or not. I did originally, when scrapped it, then made it again. My rationale for this was that I felt the manager class was doing too much without it, and making rooms a separate class allowed me to not put too much on room. I'm still not sure if this is the best way.
Describe a concept that you gained more clarity on as you worked on this assignment. | What an acceptable amount of dependency between classes.
Describe a _nominal test_ that you wrote for this assignment. | If a reservation was trying to be made before the start of another reservation
Describe an _edge case test_ that you wrote for this assignment. |  If a reservation was made on the the check-our date of another reservation and the check-in of a third
How do you feel you did in writing pseudocode first, then writing the tests and then the code? |  It was helpful, especially when I completely restarted the project three times. Keeping the same tests and slowly commenting them back in (and modifying them when needed) really helped

### CODE REVIEW

**Comment by kschumy** [comment]
Please don't grade this yet. It's really bad. Like not my usual definition of bad (i.e. it doesn't cure cancer in O(log n)) but actually bad (as in it doesn't meet the minimum requirements). I'm hoping to have it through wave three tomorrow.

**Comment by CheezItMan** [comment]
BTW, I can't run your tests because `File does not exist: /Users/chris/ada/temp/hotel/lib/reservable`

File: lib/block.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+require_relative 'reservation_dates'
+require_relative 'reservable'
+
+module Hotel
+  class Block
+    include DatesRangeModule
```
Comment (CheezItMan) [inline lib/block.rb:6]: 👍

File: lib/hotel_manager.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      @reservations << Reservation.new(res_info)
+      # return reservation_info
+    end
+
+     def get_block_rooms(id)
+      Hotel::Blocks.get_block(id) or raise ArgumentError("Invalid ID")
+     end
+
+    def burn_it_to_the_ground
+      exit
```
Comment (CheezItMan) [inline lib/hotel_manager.rb:78]: :D

File: lib/reservation_dates.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+module DatesRangeModule
+
+  # DateRange = Struct.new(:check_in, :check_out)
```
Comment (CheezItMan) [inline lib/reservation_dates.rb:3]: 👍

File: specs/block_spec.rb [hunk 4]
```diff
@@ -0,0 +1,408 @@
+require_relative 'spec_helper'
+
+describe 'Block class' do
+
+  TestDates = Struct.new(:check_in, :check_out)
+
+  describe 'Initializer' do
```
Comment (CheezItMan) [inline specs/block_spec.rb:7]: So no testing here?

File: specs/hotel_manager_spec.rb [hunk 5]
```diff
@@ -0,0 +1,356 @@
+require_relative 'spec_helper'
+
+describe 'HotelManager class' do
+
```
Comment (CheezItMan) [inline specs/hotel_manager_spec.rb:4]: No testing for creating blocks?

**Comment by CheezItMan** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility |  Check
Demonstrated loose coupling |  Check
Methods demonstrate a good use of encapsulation, inputs and outputs | Check, very good use of a mix-in
Wave 1 requirements |  Check
Wave 2 requirements |  Check
Wave 3 requirements |  Check, but I couldn't find testing of hotel admin with blocks. 
Summary | It looks like you went through a crapton of iterations here.  Just remember don't make perfect the enemy of the good.  Lots of nice work here, you hit all the learning goals, but I didn't see testing on Block for Wave 3.  Other than that excellent work!  Checkout our [reference implementation](https://github.com/droberts-ada/hotel/tree/dpr/solution).

**Pull request closed without merging**